### PR TITLE
mpd, game-music-emu: fix aarch64-darwin build

### PR DESCRIPTION
**Pull request by ylh**
###### Motivation for this change

Largely, this is to get mpd to build. Hopefully this is one small step toward ZHF.

EDIT: tacked on an mpd change I was unwittingly getting from an overlay of mine. *Now* this will ensure that mpd builds. Apologies for the extra ofborg eval.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [X] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ylh** [comment]
oh fun fact apparently this also unbreaks `ffmpeg-full` on darwin

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -20971,7 +20971,7 @@ with pkgs;
   mod_timestamp = apacheHttpdPackages.mod_timestamp;
 
   inherit (callPackages ../servers/mpd {
-    stdenv = if stdenv.cc.isClang then llvmPackages_8.stdenv else stdenv;
+    stdenv = if (with stdenv; cc.isClang && isx86_64) then llvmPackages_8.stdenv else stdenv;
```
Comment (sternenseemann) [inline pkgs/top-level/all-packages.nix:20974]: This line can be dropped completely, since #148396 will upgrade clang to 11 on all darwin platforms.

Comment (ylh) [reply pkgs/top-level/all-packages.nix:20974]: Sounds good, I'll force push to a more sensible changeset once #148396 gets merged (waiting so as not to trigger an ofborg eval where I know `x86_64-darwin` will fail).

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 147646` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages marked as broken and skipped:</summary>
  <ul>
    <li>corrscope</li>
    <li>olive-editor</li>
    <li>python39Packages.tensorflow-datasets</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>handbrake</li>
  </ul>
</details>
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>arcan.ffmpeg</li>
    <li>game-music-emu</li>
    <li>manim</li>
    <li>mpd</li>
    <li>mpd-small</li>
    <li>python310Packages.pydub</li>
    <li>python39Packages.pydub</li>
    <li>restream</li>
  </ul>
</details>

**Comment by siraben** [comment]
Thanks for fixing this on aarch64-darwin!

**Pull request merged**